frame/composable-traits/src/lending/mod.rs

```diff
@@ -7,13 +7,27 @@ use crate::{
 	defi::{CurrencyPair, DeFiEngine, MoreThanOneFixedU128},
 	time::Timestamp,
 };
-use composable_support::validation::Validate;
+use composable_support::validation::{TryIntoValidated, Validate};
 use frame_support::{pallet_prelude::*, sp_runtime::Perquintill, sp_std::vec::Vec};
 use scale_info::TypeInfo;
-use sp_runtime::{traits::One, Percent};
+use sp_runtime::{
+	traits::{One, Zero},
+	Percent,
+};
 
 use self::math::*;
 
+/// Representation for the collateral ratio of a borrower. It's possible for the borrow value to be
+/// zero when calculating this, which would result in a divide by zero error; hence the
+/// [`NoBorrowValue`][CollateralRatio::NoBorrowValue] variant.
+pub enum CollateralRatio<T> {
```

Comment:
Comments are good, so could it be Result<T, ArithmeticError as in SafeMath>? Or even option if only one Math error possible. 
Comments on method.

Problem with this enum as it is option and not composable with all other moands (result, option). So it will lead to more code.